monte_carlo.py

```diff
@@ -23,60 +27,229 @@
+                elif (bounds.get_surface_type(surface) == 0):
+                    neutron_lost = True
+                    neutron_interaction = 1
+                    neutron_distance = 0
+                    tallies["leaks"].increment(1)
+            if neutron_reflected:
+                neutron_distance -= tempd
+                
+        if not neutron_lost:
+            neutron_interaction = sample_interaction(mat)
 
+    # neutron_position turns into a vector pointing from the neutron's origin
+    # to its absorption point
+    if not neutron_lost:
+        tallies["absorptions"].increment(1)
+        if sample_fission(mat) == 1:
+            for j in xrange(sample_num_fission(mat)):
+                fission_banks["new"].add([neutron_position.x,
+                        neutron_position.y, neutron_position.z])
+                tallies["fissions"].increment(1) 
+        neutron_position.ssub([neutron_starting_point.x,
+            neutron_starting_point.y, neutron_starting_point.z])
+        tallies["crows"].increment(neutron_position.getDistance())
+        tallies["num_crows"].increment(1)
 
 '''
  @brief     Generates and transports neutron histories, calculates the mean
             crow distance
  @param     n_histories number of neutron histories to run
- @param     sigma_t total cross section of the material 
- @param     sigma_s scattering cross section of the material
+ @param     mat an instance of the Material class containing information
+            about the material
+ @param     bounds an instance of the boundaries class containing the limits
+            of the bounding box
 '''
-def generate_neutron_histories(n_histories, sigma_t, sigma_s, x_min, x_max,
-        y_min, y_max, z_min, z_max):
+def generate_neutron_histories(n_histories, mat, bounds):
     
-    crow_distances = list()
-    for i in xrange(n_histories):
-        transport_neutron(sigma_t, sigma_s, x_min, x_max, y_min, y_max, z_min,
-                z_max, crow_distances)
+    crow_distances = tally.Tally()
+    num_crow_distances = tally.Tally()
+    leakage_tally = tally.Tally()
+    absorption_tally = tally.Tally()
+    fission_tally = tally.Tally()
+    
+    tallies = {"crows": crow_distances, "num_crows": num_crow_distances,
+            "leaks": leakage_tally, "absorptions": absorption_tally,
+            "fissions": fission_tally}
 
-    mean_crow_distance = sum(crow_distances) / n_histories
-    print "Mean crow fly distance = ", mean_crow_distance
+    old_fission_locations = fission.Fission()
+    new_fission_locations = fission.Fission()
+    fission_banks = {"old": old_fission_locations,
+            "new": new_fission_locations}
+    
+    first_round = True
+    for batch in xrange(10):
+        
+        # assign the new fission locations to the old fission list
+        old_fission_locations.clear()
+        for num in xrange(new_fission_locations.length):
+            old_fission_locations.add(new_fission_locations.next())
```

Comment:
Maybe I didn't make the fission class the way you wanted me to. This doesn't work: When I used
old_fission_locations = new_fission_locations
new_fission_locations.clear()
the old fission object was empty so I received an error when I tried to sample locations from it.